worker/rhpv3.go

```diff
@@ -153,6 +153,12 @@ func (a *accounts) ForHost(hk types.PublicKey) (*account, error) {
 	return acc, nil
 }
 
+func (a *account) Balance() types.Currency {
+	a.mu.RLock()
+	defer a.mu.RUnlock()
```

Comment:
Also need to acquire `a.balanceMu`. `mu` handles deposits and withdrawals but `balanceMu` is for actually accessing the field.